src/workerd/api/sockets.c++

```diff
@@ -17,8 +17,21 @@ jsg::Ref<Socket> connectImplNoOutputLock(
   kj::Own<WorkerInterface> client = fetcher->getClient(
       ioContext, jsRequest->serializeCfBlobJson(js), "connect"_kj);
 
-  // TODO: Validate `address` is well formed. Do I need to use `parseAddress` here or is there
-  // a better way?
+  // Note that we intentionally leave it up to the connect() implementation to decide what is a
+  // valid address. This means that people using `workerd`, for example, can arrange to connect
+  // to Unix sockets (if they define a "Network" service that permits local connections, which
+  // the default internet service will not). Also, hypothetically, in W2W communications, the
+  // address could be an arbitrary string which the receiving Worker can validate however it wants.
+  //
+  // TODO(soon): This results in an "internal error" in the case that the address couldn't parse,
+  //   which is not a great experience. Should we attempt to validate the address here to give a
+  //   better error? But that takes away the backend's flexibility to define its own address
+  //   format. Maybe that's good though? It's more consistent with fetch(), which requires a
+  //   valid URL even for W2W. The only other way we can get good errors here is if we use string
+  //   matching to detect KJ's invalid-address error message, which seems pretty gross but could
+  //   work. Note that if we do decide to validate addresses, we should not try to use KJ's
+  //   `parseAddress()` but instead decide for ourselves what format we want to permit here, maybe
+  //   as a regex.
```

Comment:
Hm, I didn't consider leaving this to the `connect` implementation. But that sounds fair enough.

When I wrote this comment my primary concern was something like: `connect("blah.org HTTP/1.1\r\nARBITRARY_HTTP_HEADERS_HERE")`. Without validation this would mean the Worker can send arbitrary http headers to the proxy. But I suppose we should ensure this cannot happen in HttpClient.

Reply:
Yeah hopefully HttpClient already prevents this. If it doesn't that is a bug we need to fix.